public/app/features/alerting/unified/components/rules/central-state-history/EventDetails.tsx

```diff
@@ -15,19 +15,20 @@ import { VizWrapper } from '../../rule-editor/VizWrapper';
 import { AnnotationValue } from '../../rule-viewer/tabs/Details';
 import { LogRecord } from '../state-history/common';
 
-import { EventState } from './EventListSceneObject';
+import { EventState, FilterType } from './EventListSceneObject';
 
 interface EventDetailsProps {
   record: LogRecord;
   logRecords: LogRecord[];
+  addFilter: (key: string, value: string, type: FilterType) => void;
 }
-export function EventDetails({ record, logRecords }: EventDetailsProps) {
+export function EventDetails({ record, logRecords, addFilter }: EventDetailsProps) {
   // get the rule from the ruleUID
   const ruleUID = record.line?.ruleUID ?? '';
   const identifier = useMemo(() => {
     return parse(ruleUID, true);
   }, [ruleUID]);
-  const { error, loading, result: rule } = useCombinedRule({ ruleIdentifier: identifier });
+  const { error, loading, result: rule } = useCombinedRule({ ruleIdentifier: identifier, limitAlerts: 0 }); // we limit the alerts to 0 as we only need the rule
```

Comment:
this limit was introduced on a recent merged PR, and it will improve considerably the performance of this request